docs/proposals/note-creation-filepath-fallback-chain.md

```diff
@@ -0,0 +1,92 @@
+# Templates v2 Proposal <!-- omit in TOC -->
+
+- [Introduction](#introduction)
+- [Goals](#goals)
+- [Proposal: "New note" path resolution](#proposal-new-note-path-resolution)
+  - [Phase 1: `<workspace>/.foam/templates/new-note.md`](#phase-1-workspacefoamtemplatesnew-notemd)
+  - [`foam.rootDirectory`](#foamrootdirectory)
+  - [Phase 2: `${foam.rootDirectory ?? <home_dir>}/.foam/templates/new-note.md`](#phase-2-foamrootdirectory--home_dirfoamtemplatesnew-notemd)
+  - [Phase 3: `foam.rootDirectory`](#phase-3-foamrootdirectory)
+  - [Phase 4: Use the directory of the active editor](#phase-4-use-the-directory-of-the-active-editor)
+  - [Phase 5: Use the workspace root](#phase-5-use-the-workspace-root)
+  - [Phase 6: `foam.rootDirectory ?? <home_dir>`](#phase-6-foamrootdirectory--home_dir)
+- [Progression of Foam usage maturity](#progression-of-foam-usage-maturity)
+
+## Introduction
+
+This document aims to standardize the fallback chain that is used by Foam when determining where to create a new note.
+Currently, each command has its own fallback chain, which leads to inconsistent behaviour in Foam.
+
+This document will guide the development direction that each note-creating-command will eventually be modified to follow.
+
+Note that throughout this doc, `foam.rootDirectory` is used as a placeholder name to denote a new Foam setting that we would add.
```

Comment:
Full reply is here: https://github.com/foambubble/foam/pull/683#issuecomment-916070736

----

Q: Why is there a different defaulting behaviour for `foam.notesRootDirectory` between the "workspace" and "global" levels?
If there's a need to add a `foam` subdirectory, shouldn't it be applied at both levels for consistency? (Aside/nit: default to `notes` instead of `foam`?)